src/utils/image_loader.py

```
@@ -20,6 +20,33 @@
+# per load call and the registration is a no-op after the first success.
+_HEIF_OPENER_REGISTERED = False
+
+
+def _ensure_heif_opener() -> None:
+    """Register the HEIF/HEIC PIL opener on first use.
+
+    Called from every ``AdaptiveImageLoader`` entry point so that pi_heif
+    is imported only when the process actually decodes an image, rather
+    than eagerly at module load time.
+    """
+    global _HEIF_OPENER_REGISTERED
```

Ruff (0.15.9)

[warning] Using the global statement to update `_HEIF_OPENER_REGISTERED` is discouraged

(PLW0603)